### PR TITLE
Fix image upload request

### LINKED ISSUE #773: Updating profile image is broken

### PR DESCRIPTION
**Pull request by mradkov**
Issue was the correct headers were not passed when an image is uploaded

fixes #773

### CODE REVIEW

File: src/utils/backend.js [hunk 1]
```diff
@@ -43,11 +44,22 @@ export default class Backend {
...
+      body:
+        postParam instanceof FormData ? postParam : JSON.stringify(postParam),
+    };
+
+    Object.assign(
+      request,
+      !(postParam instanceof FormData) && {
+        headers: { 'Content-Type': 'application/json' },
+      },
+    );
```
Comment (mradkov) [inline src/utils/backend.js:59]: Here we assign the `headers` prop only in the case where no image was uploaded. 

We can optimize if we find a more clean way to have a null-safe object property assign.

Comment (davidyuk) [reply src/utils/backend.js:59]: I'm proposing to do it in a way as it was before:
```js
static sendProfileData = async (address, postParam) => backendFetch(`profile/${address}`, {
  method: 'post',
  body: postParam instanceof FormData ? postParam : JSON.stringify(postParam),
  ...!(postParam instanceof FormData) && { headers: { 'Content-Type': 'application/json' } },
});
```

File: src/utils/backend.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import { BACKEND_URL } from '../config/constants';
 import { wrapTry } from './util';
 
-const backendFetch = (path, ...args) => wrapTry(fetch(`${BACKEND_URL}/${path}`, ...args)
-  .catch((err) => console.error(err)));
+const backendFetch = (path, ...args) => wrapTry(
+  fetch(`${BACKEND_URL}/${path}`, ...args).catch((err) => console.error(err)),
+);
```
Comment (davidyuk) [inline src/utils/backend.js:6]: Would be so nice to remove `console.error(err)` from here, but it is out of the scope of this PR

File: src/utils/backend.js [hunk 3]
```diff
@@ -45,8 +46,8 @@ export default class Backend {
 
   static sendProfileData = async (address, postParam) => backendFetch(`profile/${address}`, {
     method: 'post',
-    body: JSON.stringify(postParam),
-    headers: { 'Content-Type': 'application/json' },
+    body: postParam instanceof FormData ? postParam : JSON.stringify(postParam),
+    ...!(postParam instanceof FormData) && { headers: { 'Content-Type': 'application/json' } },
```
Comment (davidyuk) [inline src/utils/backend.js:50]: we can also extract condition 🤩
```suggestion
    ...postParam instanceof FormData
      ? { body: postParam }
      : {
        body: JSON.stringify(postParam),
        headers: { 'Content-Type': 'application/json' }
      },
```
@mradkov can you squash all commits if you decide to apply this one?

Comment (mradkov) [reply src/utils/backend.js:50]: I will apply it and squash.

**Review by thepiwo — APPROVED** [review]

**Review by davidyuk — APPROVED** [review]

**Pull request merged**